packages/time-picker/src/TimePicker/utils/time.ts

```diff
@@ -25,19 +25,42 @@ export const formatDateToTime = (date: Date, locale, timeZone): string =>
     timeZone,
   }).format(date)
 
+const isNumber = (aNum: string) => {
+  if (isNaN(parseInt(aNum, 10))) {
+    return false
+  }
+  return true
+}
+
+const getDateRegEx = (segments: DateSegment[] | undefined) => {
+  let regex = ""
+  if (segments) {
+    for (const [, { type, text, isPlaceholder }] of Object.entries(segments)) {
```

Comment:
Why apply `Object.entries` on an array? 
Is there a way we write this without using a `for...of` loop as loops aren't a common practice within the codebase? 

Not super tested, but either:
```suggestion
     const regex = segments
    .filter(
      segment =>
        ((segment.type === "literal" || segment.type === "dayPeriod") &&
          segment.isPlaceholder !== undefined &&
          !segment.isPlaceholder) ||
        isNumber(segment.text)
    )
    .map(segment => segment.text)
    .join("")
```

or if you don't like multiple loops:
```suggestion
   const regex = segments.reduce((regexString, segment) => {
    if (
      ((segment.type === "literal" || segment.type === "dayPeriod") &&
        segment.isPlaceholder !== undefined &&
        !segment.isPlaceholder) ||
      isNumber(segment.text)
    ) {
      regexString += segment.text
    }
    return regexString
  }, "")

```

Reply:
Oh, I initially wanted the index here but I can just for..of the object. I think simple javascript is well understood by everyone.